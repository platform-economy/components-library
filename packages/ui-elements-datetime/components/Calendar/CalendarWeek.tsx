import * as React from 'react';
import classnames from 'classnames';

export type CalendarWeekProps = {
  weekNumber: number;
  isCurrent: boolean;
};

export class CalendarWeek extends React.Component<CalendarWeekProps> {
  render() {
    const { isCurrent, children } = this.props;
    return (
      <tr
        className={classnames('CalendarWeek', {
          current: isCurrent,
        })}
      >
        {children}
      </tr>
    );
  }
}
