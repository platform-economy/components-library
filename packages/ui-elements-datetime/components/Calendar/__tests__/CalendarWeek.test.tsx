import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import 'jest-styled-components';
import { CalendarWeek } from '../CalendarWeek';

describe('ui-elements-datetime', () => {
  describe('Components', () => {
    describe('CalendarDay', () => {
      [false, true].forEach((isCurrent) => {
        it(`should render: isCurrent=${isCurrent}`, () => {
          const tree = TestRenderer
            .create(<CalendarWeek weekNumber={1} isCurrent={isCurrent}/>)
            .toJSON();
          expect(tree).toMatchSnapshot();
        });
      });
    });
  });
});
