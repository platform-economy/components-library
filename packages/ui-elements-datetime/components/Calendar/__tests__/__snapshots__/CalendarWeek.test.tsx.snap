// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui-elements-datetime Components CalendarDay should render: isCurrent=false 1`] = `
<tr
  className="CalendarWeek"
/>
`;

exports[`ui-elements-datetime Components CalendarDay should render: isCurrent=true 1`] = `
<tr
  className="CalendarWeek current"
/>
`;
